experiment/cherrypicker/main.go

```diff
@@ -0,0 +1,167 @@
+	}
+
+	webhookSecretRaw, err := ioutil.ReadFile(*webhookSecretFile)
+	if err != nil {
+		logrus.WithError(err).Fatal("Could not read webhook secret file.")
+	}
+	webhookSecret := bytes.TrimSpace(webhookSecretRaw)
+
+	oauthSecretRaw, err := ioutil.ReadFile(*githubTokenFile)
+	if err != nil {
+		logrus.WithError(err).Fatal("Could not read oauth secret file.")
+	}
+	oauthSecret := string(bytes.TrimSpace(oauthSecretRaw))
+
+	_, err = url.Parse(*githubEndpoint)
+	if err != nil {
+		logrus.WithError(err).Fatal("Must specify a valid --github-endpoint URL.")
+	}
+
+	githubClient := github.NewClient(oauthSecret, *githubEndpoint)
+	if *dryRun {
+		githubClient = github.NewDryRunClient(oauthSecret, *githubEndpoint)
+	}
+
+	gitClient, err := git.NewClient()
+	if err != nil {
+		logrus.WithError(err).Fatal("Error getting git client.")
+	}
+
+	logger := logrus.StandardLogger()
+	// TODO: Use global option from the prow config.
+	logrus.SetLevel(logrus.DebugLevel)
+	githubClient.Logger = logger.WithField("client", "github")
+	gitClient.Logger = logger.WithField("client", "git")
+
+	// The bot name is used to determine to what fork we can push cherry-pick branches.
+	botName, err := githubClient.BotName()
+	if err != nil {
+		logrus.WithError(err).Fatal("Error getting bot name.")
+	}
+
+	repos, err := githubClient.GetRepos(botName, true)
+	if err != nil {
+		logrus.WithError(err).Fatal("Error listing bot repositories.")
+	}
+	var exists bool
+	for _, r := range repos {
+		if !r.Fork {
+			continue
+		}
+		if r.FullName == botName+"/"+repo {
+			exists = true
+			break
+		}
+	}
+	// Fork repo if it doesn't exist.
+	if !exists {
+		if err := githubClient.CreateFork(org, repo); err != nil {
+			logrus.WithError(err).Fatalf("Error while forking %s/%s.", org, repo)
+		}
+		// Wait for at most 5 minutes for the fork to appear on Github.
+		after := time.After(5 * time.Minute)
+		tick := time.Tick(5 * time.Second)
+		forkURL := fmt.Sprintf("https://github.com/%s/%s", botName, repo)
+	out:
+		for {
+			select {
+			case <-tick:
+				resp, err := http.Get(forkURL)
+				if err == nil && resp.StatusCode == 200 {
+					resp.Body.Close()
+					break out
+				}
+				if err == nil {
+					resp.Body.Close()
+				}
+			case <-after:
+				logrus.Fatalf("Failed to get newly created fork of %s/%s for %s after 5m - contact Github at https://github.com/contact", org, repo, botName)
+			}
+		}
+	}
+
+	server := &Server{
+		BotCredentials: oauthSecret,
+		BotName:        botName,
+		GitClient:      gitClient,
+		GithubClient:   githubClient,
+		HMACSecret:     webhookSecret,
+		Logger:         logrus.StandardLogger().WithField("client", "cherrypicker"),
+	}
+
+	// Return 200 on / for health checks.
+	http.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {})
+	// For /hook, handle a webhook normally.
+	http.Handle("/hook", server)
+	logrus.Fatal(http.ListenAndServe(":"+strconv.Itoa(*port), nil))
+}
+
+func validateRepo(r string) (owner, repo string, err error) {
```

Comment:
Could just ask for `--org` and `--repo` if we're just parsing them out first thing anyway

Reply:
I am thinking of removing the `--repo` option altogether. This should flow from the plugins config.